x-pack/plugins/enterprise_search/public/applications/shared/use_did_update_effect/use_did_update_effect.tsx

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * Sometimes we don't want to fire the initial useEffect call.
+ * This custom Hook only fires after the intial render has completed.
+ */
+import { useEffect, useRef } from 'react';
+
+type TInput = string | number | object | Function | boolean | null | undefined;
```

Comment:
Could we use type from React? Looks like DependencyList is what we need from this file: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L891

Or, if we can't import the type, maybe we could use `inputs: ReadonlyArray<unknown>` as they do it?

Reply:
Great catch!!